source/tools/glkernel-cmd/main.cpp

```diff
@@ -0,0 +1,58 @@
+#include "KernelGeneration.h"
+#include "KernelToJson.h"
+
+#include <iostream>
+#include <fstream>
+
+#include <cppassist/cmdline/ArgumentParser.h>
+
+#include <cppexpose/variant/Variant.h>
+#include <cppexpose/json/JSON.h>
+
+int main(int argc, char* argv[])
+{
+    cppassist::ArgumentParser argParser;
+    argParser.parse(argc, argv);
```

Comment:
We already have a more sophisticated framework for parsing command line parameters in cppassist: CommandLineProgram. I suggest to use that for real command line programs like this one, it helps with specifying the possible options and handles such things as outputting help texts automatically. There is also an example for this is cppassist.